lib/metadata/hfbMetadata.hpp

```diff
@@ -29,6 +26,12 @@ struct hfbMetadata {
     uint32_t num_samples_expected;
     /// Data size of frame after compression
     uint32_t compressed_data_size;
+    /// The number of beams in the data.
+    uint32_t num_beams;
+    /// The number of sub-frequencies in the data.
+    uint32_t num_subfreq;
+    /// ID of the dataset
+    dset_id_t dataset_id;
 };
 
 // Helper functions to save lots of pointer work
```

Comment:
Do you still need all these C style getters and setters? I think with the FrameView you probably don't need them and could just remove them.

Reply:
So I use these in `integrateHFBData.cpp` instead of directly accessing the metadata:

https://github.com/kotekan/kotekan/blob/ca1a367a7e8fcd55aac1117df7bd62eafdc5ec77/lib/stages/integrateHFBData.cpp#L201-L224